odoo/models.py

```diff
@@ -2665,8 +2665,22 @@ def add(name, definition):
             if not getattr(cls, 'pool', None)
             for constraint in getattr(cls, '_local_sql_constraints', ())
         }
-
-        for (key, definition, _) in self._sql_constraints:
+        sql_constraints = list(self._sql_constraints)
+
+        if self._parent_store:
+            # we need this extension
+            cr.execute('CREATE EXTENSION IF NOT EXISTS btree_gist')
```

Comment:
For production use we recommend running with a non-super-user PG role, thus `CREATE EXTENSION` may not be allowed. We should guard against that and issue a warning in the log.
PG does not seem to raise an error if the user does not have `CREATE EXTENSION` privileges but the extension already exists and `IF NOT EXISTS` is specified, which makes it a little bit easier to handle.